packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -1230,6 +1233,9 @@ class _AppRunLogger extends Logger {
 
   @override
   bool get hasTerminal => false;
+
+  @override
+  void clear() => parent?.printStatus(globals.terminal.clearScreen() + '\n');
```

Comment:
This does not have a terminal attached, see `hasTerminal` so you cannot use this here. Instead, no-op